////////////////////////////////////////////////////////////////////////////////////////////////////
///
/// @file      Kepler.proto
///
/// @brief     This header file describes the protocol buffer messages associated with 
///            the Kepler remote configuration.
///
/// @attention Copyright (C) 2018 Bose Corporation All Rights Reserved
///
///            Bose Corporation
///            The Mountain Road,
///            Framingham, MA 01701-9168
///            U.S.A.
///
///            This program may not be reproduced, in whole or in part, in any form by any means
///            whatsoever without the written permission of Bose Corporation.
///
////////////////////////////////////////////////////////////////////////////////////////////////////

import "ProductSTS.proto";
import "SystemSourcesProperties.proto";

package KeplerPb;

message KeplerConfig
{
    // "Source" is a value synthesized from various pieces of information
    // about the system state
    enum Source
    {
        // "fixed" product sources
        SOUNDTOUCH          = 1;
        TV                  = 2;
        BLUETOOTH           = 3;
        // configurable (SLOT) sources (type determined from activation key)
        GAME                = 4;
        STREAMING           = 5;
        CBL_SAT             = 6;
        BD_DVD              = 7;
        // setup sources
        ADAPTIQ             = 8;
        ACCESSORY_PAIRING   = 9;
        OOB                 = 10;
        // we are a zone member
        ZONE                = 11;
        // we are powered off
        OFF                 = 12;
    }
    
    enum SourceKey
    {
        SOUNDTOUCH_KEY      = 1;
        TV_KEY              = 2;
        BLUETOOTH_KEY       = 3;
        GAME_KEY            = 4;
        CBL_SAT_KEY         = 5;
        BD_DVD_KEY          = 6;
    }

    message BacklightEntry
    {
        // zones are lit unconditionally
        // zonesUnconfigured are lit when the source is configured
        // zonesConfigured are lit when the source is not configured
        optional Source                                     source              = 1;         
        repeated uint32                                     zones               = 2;
        repeated uint32                                     zonesUnconfigured   = 3;
        repeated uint32                                     zonesConfigured     = 4;

        // The following are used to match this entry to a CAPS source
        // deviceType is only used for SLOT sources
        optional string                                     sourceName          = 5;
        optional string                                     sourceAccountName   = 6;
        optional SystemSourcesProperties.DEVICE_TYPE_       deviceType          = 7;        
    }

    repeated BacklightEntry backlightConfig             = 1;

}

////////////////////////////////////////////////////////////////////////////////////////////////////
///                                        End of File                                           ///
////////////////////////////////////////////////////////////////////////////////////////////////////
